packages/rest/src/lib/utils/utils.ts

```diff
@@ -43,7 +43,7 @@ function serializeSearchParam(value: unknown): string | null {
  * @param options - The options to use
  * @returns A populated URLSearchParams instance
  */
-export function makeURLSearchParams(options?: Record<string, unknown>) {
+export function makeURLSearchParams<T extends object & {}>(options?: Readonly<T>) {
```

Comment:
from ts 4.7 (or 8 ig) `T & {}` is `{}`. So this will become `object`. So do we need this intersection?

Reply:
Yes.

- `object` means any object value.
- `{}` means any non-nullish value (`unknown` sans `undefined | null`).

When combined, `object & {}`, it means that it's any non-nullish object.

Reply:
check https://tsplay.dev/wE4D2W
when using `object`, it's already non-nullish and adding the intersection of `{}` with any value returns the same value like 
`number & {}` is also `number`

Reply:
Besides `type C = Test<null>` needing to be `type C = TestWithInt<null>` on line 6, yes, you're right.

I'll remove `{}` right away, thank you for letting me know, I didn't know the semantics changed in 4.8 (I checked the announcements for [4.7](https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/), [4.8](https://devblogs.microsoft.com/typescript/announcing-typescript-4-8/), and [4.9](https://devblogs.microsoft.com/typescript/announcing-typescript-4-9/), but only found [Unconstrained Generics No Longer Assignable to `{}`](https://devblogs.microsoft.com/typescript/announcing-typescript-4-8/#unconstrained-generics-no-longer-assignable-to) inside the 4.8 announcement, it was a bit hidden to be fair). 🙏🏼